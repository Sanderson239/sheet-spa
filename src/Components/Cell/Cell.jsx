import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'semantic-ui-react';
import { connect } from 'react-redux';


export const Cell = ({ cellId, cellsById }) => (
  <Table.Cell>
    {`${cellsById[cellId]}`}
  </Table.Cell>
);


Cell.propTypes = {
  cellId: PropTypes.number.isRequired,
  cellsById: PropTypes.shape({ 1: PropTypes.string }).isRequired,
};

const mapStateToProps = (state, ownProps) => {
  const cellIds = state.sheet.rowsById[ownProps.rowId].cellIds;
  const cellsById = state.sheet.rowsById[ownProps.rowId].cellsById;
  return {
    cellIds,
    cellsById,
  };
};


export default connect(
  mapStateToProps,
)(Cell);
