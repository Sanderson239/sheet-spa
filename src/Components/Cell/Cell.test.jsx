import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Cell from './Cell';

describe('Spreadsheet cell', () => {
  it('should render a cell', ({ cellId, cellsById }) => {
    const component = shallow(<Cell />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
