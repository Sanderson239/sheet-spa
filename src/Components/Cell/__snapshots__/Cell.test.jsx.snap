// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<td
  className=""
>
  A2
</td>
`;
